packages/api/cms-api/src/user-permissions/content-scope.service.ts

```diff
@@ -21,63 +21,49 @@ export class ContentScopeService {
         return isEqual({ ...scope1 }, { ...scope2 });
     }
 
-    async inferScopeFromExecutionContext(context: ExecutionContext): Promise<ContentScope | ContentScope[] | undefined> {
+    async inferScopesFromExecutionContext(context: ExecutionContext): Promise<ContentScope[]> {
+        const contentScopes: ContentScope[] = [];
         const args = await this.getArgs(context);
 
-        const affectedEntity = this.reflector.getAllAndOverride<AffectedEntityMeta>("affectedEntity", [context.getHandler(), context.getClass()]);
-        if (affectedEntity) {
-            let contentScope: ContentScope | ContentScope[] | undefined;
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            const repo = this.orm.em.getRepository<any>(affectedEntity.entity);
+        const affectedEntities = this.reflector.getAllAndOverride<AffectedEntityMeta>("affectedEntities", [context.getHandler()]) || [];
+        for (const affectedEntity of affectedEntities) {
             if (affectedEntity.options.idArg) {
-                if (!args[affectedEntity.options.idArg]) {
-                    throw new Error(`${affectedEntity.options.idArg} arg not found`);
-                }
-                const row = await repo.findOneOrFail(args[affectedEntity.options.idArg]);
-                if (row.scope) {
-                    contentScope = row.scope;
-                } else {
-                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                    const scoped = this.reflector.getAllAndOverride<ScopedEntityMeta>("scopedEntity", [
-                        affectedEntity.entity as EntityClass<unknown>,
-                    ]);
-                    if (!scoped) {
-                        return undefined;
+                if (!args[affectedEntity.options.idArg]) throw new Error(`${affectedEntity.options.idArg} arg not found`);
+                const repo = this.orm.em.getRepository<{ scope?: ContentScope }>(affectedEntity.entity);
+                const id = args[affectedEntity.options.idArg];
+                const ids = Array.isArray(id) ? id : [id];
+                for (const id of ids) {
+                    const row = await repo.findOneOrFail(id);
+                    if (row.scope) {
+                        contentScopes.push(row.scope);
```

Comment:
Note: This could add the same scope multiple times, but I believe it is okay to ignore